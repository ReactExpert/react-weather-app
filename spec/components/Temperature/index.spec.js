import React from 'react';
import Temperature from '../../../src/components/Temperature';
import { enzymeConfig, mount } from '../../enzymeConfig';

enzymeConfig();

describe('Temperature', () => {
  it('matches snapshot', () => {
    const component = mount(<Temperature weather='clear' temperature={10} />);

    expect(component).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('calls animate', (done) => {
      const component = mount(<Temperature weather='clear' temperature={10} />);
      const instance = component.instance();

      instance.animate = jest.fn();

      setTimeout(() => {
        expect(instance.animate).toBeCalled();
        done();
      }, 500);
    });
  });
});
