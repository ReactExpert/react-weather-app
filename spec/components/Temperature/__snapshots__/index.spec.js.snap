// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Temperature matches snapshot 1`] = `
<Temperature
  temperature={10}
  weather="clear"
>
  <section
    className="current-condition"
  >
    <div
      className="wrapper-temperature"
    >
      <span
        className="temperature__status"
      >
        clear
      </span>
      <div
        className="temperature"
      >
        <span
          className="temperature__value"
        >
          10
        </span>
        <div
          className="temperature__unit"
        >
          <span
            className="temperature__unit-dot"
          />
          <span
            className="temperature__unit-letter"
          >
            c
          </span>
        </div>
      </div>
    </div>
  </section>
</Temperature>
`;
